### PR TITLE
Refactor redux of account - Closes #2021

### LINKED ISSUE #2021: Review and refactor the actions, reducers and middleware for accounts.
### Descriptions
This is a child ticket for #2020 .
With a short review over the code, we can find several redundant or obsolete actions. accordingly there are cases defined in the reducers switch-case statement which can be refactored or even removed. 
The naming of the actions seem to need revision.

### PR DESCRIPTION
**Pull request by slaweet**
### What issue have I solved?

#2021

### How have I implemented/fixed it?

The PR contained at first commits (26a89a917229fc6c8fcd0bf5d259225117f67b8c cba16acf7c68a3f92ad9e9ecbe35729a77bb25f6) with TODO comments that describe what should be refactored.

Changes made include:
- [x] Remove unused `accountAddVotes` and `isDelegate` from account reducer
- [x] Remove unnecessary code from account middleware
- [x] Migrate account reducer and actions test to Jest assertions
- [x] Add jsdoc comments to actions that are kept
- [x] Simplify login action

The only left TODO comment is to remove `secondPassphraseRegistered` action and use `withSignAndBroadcast.js HOC` outlined in https://github.com/LiskHQ/lisk-hub/commit/2222c4a1fb0eb41a1d5cf590f6dd573c231ec742 Since it is a more fundamental change that affects all pages that send a transaction and needs more time to review by everyone, I suggest not to block merging this PR by it. 

### How has this been tested?

- Check login, and wallet page still works. Check all tests pass.

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)

### CODE REVIEW

File: src/actions/account.js [hunk 1]
```diff
@@ -190,7 +192,15 @@ export const updateAccountDelegateStats = account =>
...
+/**
+ * This action is used on login to fetch account info for all enabled token
+ *
+ * @param {Object} data - for hardware wallets it contains publicKey and hwInfo,
+ *    otherwise contains passphrase
+ * @param {String} passphrase - BIP39 passphrase of the account
+ * @param {String} publicKey - Lisk publicKey used for hardware wallet login
+ * @param {Object} hwInfo - info about hardware wallet we're trying to login to
+ */
 export const login = ({ passphrase, publicKey, hwInfo }) => async (dispatch, getState) => {
```
Comment (osvaldovega) [inline src/actions/account.js]: @slaweet I checked the comments and all looks good, in the other hand jus as a comment I think this login action can be improve, so don't know if that will be part of this PR or another PR

Comment (slaweet) [reply src/actions/account.js]: @osvaldovega improving login action can be part of this PR if it will keep the same functionality. What do you have in mind?

One separate improvement of login I'm aware of is https://github.com/LiskHQ/lisk-hub/issues/2197 but, but so far it is missing specification of what exactly should happen with BTC in that case.
